clinica/pipelines/t1_linear/t1_linear_pipeline.py

```diff
@@ -0,0 +1,258 @@
+
+    Args:
+        bids_directory (str): Folder with BIDS structure.
+        caps_directory (str): Folder where CAPS structure will be stored.
+        ref_template (str): reference template used for image registration.
+        working_directory (str): Folder containing a temporary space to save.
+        intermediate results.
+        tsv: The Subjects-Sessions list file (in .tsv format).
+
+    Returns:
+        A clinica pipeline object containing the T1 Linear pipeline.
+
+    Raises:
+
+    Example:
+        >>> from t1_linear import T1Linear
+        >>> pipeline = T1Linear('~/MYDATASET_BIDS', '~/MYDATASET_CAPS')
+        >>> pipeline.parameters = {
+        >>>     # ...
+        >>> }
+        >>> pipeline.base_dir = '/tmp/'
+        >>> pipeline.run()
+    """
+
+    def check_custom_dependencies(self):
+        """Check dependencies that can not be listed in the `info.json` file.
+        """
+        pass
+
+    def get_input_fields(self):
+        """Specify the list of possible inputs of this pipeline.
+
+        Returns:
+            A list of (string) input fields name.
+        """
+
+        return ['t1w']
+
+    def get_output_fields(self):
+        """Specify the list of possible outputs of this pipeline.
+
+        Returns:
+            A list of (string) output fields name.
+        """
+
+        return ['image_id']
+
+    def build_input_node(self):
+        """Build and connect an input node to the pipeline.
+        """
+        from os import pardir
+        from os.path import dirname, join, abspath, split, exists
+        import sys
+        import nipype.interfaces.utility as nutil
+        import nipype.pipeline.engine as npe
+        from clinica.utils.inputs import check_bids_folder
+        from clinica.utils.participant import get_subject_session_list
+        from clinica.utils.exceptions import ClinicaBIDSError, ClinicaException
+        from clinica.utils.inputs import clinica_file_reader
+        from clinica.utils.input_files import T1W_NII
+        from clinica.utils.inputs import fetch_file
+        from clinica.utils.stream import cprint
+
+        check_bids_folder(self.bids_directory)
+        is_bids_dir = True
+
+        root = dirname(abspath(join(abspath(__file__), pardir, pardir)))
+        path_to_mask = join(root, 'resources', 'masks')
+        self.ref_template = join(
+                path_to_mask, 'mni_icbm152_t1_tal_nlin_sym_09c.nii')
+        self.ref_crop = join(path_to_mask, 'ref_cropped_template.nii.gz')
+        url1 = "https://aramislab.paris.inria.fr/files/data/img_t1_linear/ref_cropped_template.nii.gz"
+        url2 = "https://aramislab.paris.inria.fr/files/data/img_t1_linear/mni_icbm152_t1_tal_nlin_sym_09c.nii"
+
+        if not(exists(self.ref_template)):
+            try:
+                fetch_file(url2, self.ref_template)
+            except IOError as err:
+                cprint('Unable to download required template (mni_icbm152) for processing:', err)
```

Comment:
Why the exception is not raised?

Reply:
How do you tested ? (You tried to generate an `IOError`?)

Reply:
From what I read on your, if a `IOError` is caught, only the `cprint` is displayed meaning that `self.ref_template` will not contain :thinking: 

I had no issue to run your pipeline for information :) but I don't know how a `IOError` could happen (lack of space) and was wondering how to handle this.

Anyway, this was a minor question.